### PR TITLE
Improvements to truss CLI.

### PR DESCRIPTION
**Pull request by squidarth**
# Summary

Made some improvements to the `truss` CLI:

* `truss predict` now supports `--model` and `--model-version` arguments
* `truss configure` now exists which edits the `.trussrc` file

Will follow up this change with a couple more changes:
1. Printing URL after truss push 
2. Clean up the get_baseten_service path -- the code here is quite messy (I wish we could use pattern matching!)
3. I really dislike that we edit the config.yaml after truss push, it feels pretty janky that we do that and would like to revisit that

# testing

Truss Push:

```
$ poetry run truss push
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model Baseten Test Model was successfully pushed ✨

|---------------------------------------------------------------------------------------|
| Your model has been deployed as a draft. Draft models allow you to                    |
| iterate quickly during the deployment process.                                        |
|                                                                                       |
| When you are ready to publish your deployed model as a new version,                   |
| pass `--publish` to the `truss push` command. To monitor changes to your model and    |
| rapidly iterate, run the `truss watch` command                                        |
|                                                                                       |
|---------------------------------------------------------------------------------------|

🪵  View logs for your deployment at https://app.baseten.co/models/pBDEAQ0/versions/wpjvdlw/logs
? 🗂  Open logs in a new tab? Yes
```

Truss predict:

```
@squidarth ➜ /workspaces/truss/truss-public-gh-repo-test (main) $ poetry run truss predict -d '{"inputs": "b"}'
{
  "predictions": "b"
}

@squidarth ➜ /workspaces/truss/truss-public-gh-repo-test (main) $ poetry run truss predict -d '{"inputs": "b"}' --model pBDEAQ0
{
  "predictions": "b"
}

@squidarth ➜ /workspaces/truss/truss-public-gh-repo-test (main) $ poetry run truss predict -d '{"inputs": "b"}' --model-version wpjvdlw
{
  "predictions": "b"
}

```

### CODE REVIEW

File: truss/remote/baseten/service.py [hunk 1]
```diff
@@ -29,7 +29,12 @@ def is_live(self) -> bool:
     def is_ready(self) -> bool:
         raise NotImplementedError
 
-    def predict(self, model_request_body: Dict):
+    def predict(
+        self,
+        model_request_body: Dict,
+        model_id: Optional[str] = None,
```
Comment (bolasim) [inline truss/remote/baseten/service.py:35]: what's the point of adding `model_id` and model_version_id` here if they're unused?

File: truss/remote/baseten/service.py [hunk 2]
```diff
@@ -29,7 +29,12 @@ def is_live(self) -> bool:
...
+        model_version_id: Optional[str] = None,
+    ):
```
Comment (bolasim) [inline truss/remote/baseten/service.py:37]: Shouldn't we change the `_service_url` based on if you wanna invoke by model_id or model_version_id?

**Review by bolasim — COMMENTED** [review]
I am confused on how the predict actually works if we're not changing the url we're calling based on the model identifier.

File: truss/remote/baseten/service.py (hunk 2 above)
Comment (squidarth) [reply truss/remote/baseten/service.py:51]: I wasn't sure what the best way to handle this was. It felt weird to have to have two places where we process the `ModelIdentifier` (constructing the `BasetenService` AND in the `predict` function), so the way this works now is that the model_version_id gets resolved here in the baseten service, regardless of what you pass (model id, version id, or name), and we always invoke on that.

I can change predict to take a `ModelIdentifier` as well, but this felt kinda messy.

Comment (bolasim) [reply truss/remote/baseten/service.py:51]: why not just the `_service_url` based on the constructor of the BasetenService?

File: truss/cli/cli.py [hunk 3]
```diff
@@ -377,9 +423,45 @@ def push(
...
-    _ = remote_provider.push(tr, model_name, publish=publish, trusted=trusted)  # type: ignore
+    service = remote_provider.push(tr, model_name, publish=publish, trusted=trusted)  # type: ignore
+
+    model_overview_url = service.resource_url
+    click.echo(f"✨ Model {model_name} was successfully pushed ✨")
 
-    click.echo(f"Model {model_name} was successfully pushed.")
+    if not publish:
+        draft_model_text = """
+|---------------------------------------------------------------------------------------|
```
Comment (squidarth) [inline truss/cli/cli.py:433]: adding more information after `truss push`. we can iterate on the copy here.

Comment (bolasim) [reply truss/cli/cli.py:434]: should we offer to open this page like we do with the `truss watch`?

File: truss/remote/baseten/service.py (hunk 2 above)
Comment (squidarth) [reply truss/remote/baseten/service.py:53]: cool -- fixed!

File: truss/cli/cli.py (hunk 3 above)
Comment (squidarth) [reply truss/cli/cli.py:433]: Sure -- updated

**Comment by squidarth** [comment]
Thanks for review @bolasim , cleaned some of this up, lmk what you think.

File: truss/remote/baseten/remote.py [hunk 4]
```diff
@@ -68,44 +74,74 @@ def push(  # type: ignore
             model_version_id=model_version_id,
             is_draft=not publish,
             api_key=self._auth_service.authenticate().value,
-            service_url=f"{self._remote_url}/model_versions/{model_version_id}",
+            service_url=f"{self._remote_url}/model_versoins/{model_version_id}",
```
Comment (bolasim) [inline truss/remote/baseten/remote.py:77]: typo? I think this might break things

**Review by bolasim — APPROVED** [review]
LGTM. Thanks for the PR. There is only minor typo to fix before merging please.

**Comment by squidarth** [comment]
going to follow this up w/ some tests

**Pull request merged**